geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/functions/GatewaySenderCreateFunction.java

```diff
@@ -122,10 +122,12 @@ private GatewaySender createGatewaySender(Cache cache,
     }
 
     Integer dispatcherThreads = gatewaySenderCreateArgs.getDispatcherThreads();
-    if (dispatcherThreads != null && dispatcherThreads > 1) {
+    if (dispatcherThreads != null) {
       gateway.setDispatcherThreads(dispatcherThreads);
+    }
 
-      String orderPolicy = gatewaySenderCreateArgs.getOrderPolicy();
+    String orderPolicy = gatewaySenderCreateArgs.getOrderPolicy();
+    if (dispatcherThreads != null && dispatcherThreads > 1) {
```

Comment:
Is the idea here that if dispatcher threads is 1, the order policy doesn't really matter, otherwise we should remove this dispatcherThreads > 1 check too correct?

Reply:
Yes, If we have 1 dispatcher thread we don't need order policy. In case there is a more dispatcher threads than 1, order policy is needed.

Reply:
Thats not what it looks like from "GatewaySenderFactoryImpl". The order policy is required if its using more than 1 dispatcher thread and if its not provided, it gets set to default one. 
And one could set order policy without setting the dispatcher thread. This is from the GatewaySenderFactoryImpl; I haven't looked into how it is used during event dispatch.

Based on logic in GatewaySenderFactoryImpl,  this could be
if (dispatcherThreads != null) 
 gateway.setDispatcherThreads(dispatcherThreads);

if (orderPolicy != null) {
try {
OrderPolicy.valueOf(orderPolicy) // orderPolicy could be null or invalid string
gateway.setOrderPolicy(OrderPolicy.valueOf(orderPolicy));
catch (Exception) {
// Throw an error message to user?
}
}

Reply:
Hi @agingade, 
OrderPolicy cannot have an invalid string as it doesn't allow any string except this from the OrderPolicy.ENUM: KEY, THREAD and PARTITION. 

1. If dispatcher threads is more than 1 you need to set order-policy to one of this OrderPolicy.ENUM. In case you set any other string you will get:
_java.lang.IllegalArgumentException: Failed to convert 'XXX' to type OrderPolicy for option 'order-policy'
No enum constant org.apache.geode.cache.wan.GatewaySender.OrderPolicy.XXX_
2. If dispatcher threads is equal 1 it doesn't need order-policy as it is alone there(You can set order-policy to some of this values, but they don't have any impact as it will not be used). 
3. If it is lower than 1 you will get exception and message that dispatcher threads cannot be less than 1.